src/version.cpp

```diff
@@ -63,16 +63,15 @@ std::string Exiv2::versionNumberHexString()
 
 static bool shouldOutput(const exv_grep_keys_t& greps,const char* key,const std::string& value)
 {
-    bool bPrint = greps.empty();
-    for (auto g = greps.begin(); !bPrint && g != greps.end(); ++g) {
+    if (greps.empty())
+        return true;
+
+    return std::any_of(greps.begin(), greps.end(), [key, &value](const std::regex& g) {
```

Comment:
The centos7 build is complaining about this:

```cpp
/__w/exiv2/exiv2/src/version.cpp:69:72: error: 'regex' in namespace 'std' does not name a type
     return std::any_of(greps.begin(), greps.end(), [key, &value](const std::regex& g) {
```

It looks like what it is missing is the inclusion of `#include <regex>` at the top of the file.

Reply:
Ugh CentOS's GCC has quite broken C++11 support. Hopefully that fixes it.

Reply:
Turns out it's related to https://github.com/Exiv2/exiv2/commit/309998f9d432a7569db314f80bd03eae80fe8b09